ostd/src/arch/x86/cpu/context/mod.rs

```diff
@@ -480,6 +493,19 @@ impl XSaveArea {
             xsave_area.assume_init()
         }
     }
+
+    /// Returns the byte offset of the PKRU register in the extended state area.
+    fn ext_state_pkru_offset() -> usize {
+        // From Intel SDM Vol. 1 Section 13.5.7, `CPUID.(EAX=0DH,ECX=9):EBX`
+        // enumerates the offset.
+        let cpuid = cpuid::cpuid!(0x0D, 0x9);
+        let offset = cpuid.ebx as usize;
```

Comment:
How about saving this offset into a static atomic variable on the first call so that you don't have to call CPUID subsequently?